### PR TITLE
[WIP] REPL shell mode for windows

### PR DESCRIPTION
**Pull request by GregPlowman**
Take 2 for #20776
(I made a mistake with GitHub when trying to modify that PR, sorry)

Summary:
- `JULIA_SHELL` can be set to:
  - cmd
  - powershell
  - busybox
  - "" for no shell, execute command directly (current behavior) 

- `cmd` is default for Windows users.

- For Windows, the raw line is processed, rather than converting to `Cmd` and then re-assembling into a Windows command line. Windows shell commands are passed as a string anyway, so this gives the user more control over any escaping. I think it works quite well.

- Modified special handling of `cd`.  When there are 2 or more arguments, command line is processed as normal, rather than throwing an error. This allows commands such as `cd $dir && $command` to be executed.

### CODE REVIEW

File: base/client.jl [hunk 1]
```diff
@@ -82,17 +82,22 @@ answer_color() = text_colors[repl_color("JULIA_ANSWER_COLOR", default_color_answ
 stackframe_lineinfo_color() = repl_color("JULIA_STACKFRAME_LINEINFO_COLOR", :bold)
 stackframe_function_color() = repl_color("JULIA_STACKFRAME_FUNCTION_COLOR", :bold)
 
-function repl_cmd(cmd, out)
-    shell = shell_split(get(ENV,"JULIA_SHELL",get(ENV,"SHELL","/bin/sh")))
-    shell_name = Base.basename(shell[1])
+function repl_cmd(line::AbstractString, out)
+    if is_windows()
```
Comment (ararslan) [inline base/client.jl]: Should use `@static` when conditioning on the OS inside a function

Comment (tkelman) [inline base/client.jl]: not unless you're emitting code that doesn't compile on multiple platforms, or you really need to eliminate the comparison for runtime performance. better to leave the static out if neither is the case, have all branches at least get compiled.

Comment (GregPlowman) [inline base/client.jl]: I think on v0.6, code is conditionally compiled with and without `@static`
So maybe there's no runtime performance difference?

Both `test1()` and `test2()` produce the same llvm:
```
test1() = is_windows() ? 1 : "a"
test2() = @static is_windows() ? 1 : "a"
```
```
julia> @code_llvm test1()

; Function Attrs: uwtable
define i64 @julia_test1_68517() #0 !dbg !5 {
top:
  ret i64 1
}

julia> @code_llvm test2()

; Function Attrs: uwtable
define i64 @julia_test2_68518() #0 !dbg !5 {
top:
  ret i64 1
}
```

**Comment by stevengj** [comment]
Bump — would be nice to see this merged in some form.

**Comment by GregPlowman** [comment]
Replaced by #23703

**Pull request closed without merging**